spinnman/transceiver.py

```diff
@@ -2859,12 +2859,12 @@ def register_udp_listener(self, callback, connection_class,
         if connection is None:
             connection = connection_class(local_host=local_host)
             self._all_connections.add(connection)
-        listener = ConnectionListener(connection)
-        listener.start()
-        self._udp_receive_connections_by_port[connection.local_port][
-            local_host] = (connection, listener)
-        connections_of_class.append((connection, listener))
-        listener.add_callback(callback)
+            listener = ConnectionListener(connection)
+            listener.start()
+            self._udp_receive_connections_by_port[connection.local_port][
+                local_host] = (connection, listener)
+            connections_of_class.append((connection, listener))
+            listener.add_callback(callback)
```

Comment:
My guess would be that this line could be unindented.  That would mean that you add the callback to an existing listener...